[C++] v4.1 release (#129)
mongodb/docs-cpp: source/whats-new.txt

The v4.1 driver release includes the following new features:

> - Replaces the ``storage_options()`` method with ``storage_engine()`` in ``mongocxx::v_noabi::options::index``

Reviewer: ```suggestion
- Deprecates ``storage_options()`` method for the newly added ``storage_engine()`` method in ``mongocxx::v_noabi::options::index``
```

Suggest avoiding "Replaces" since `storage_options()` is not-yet removed (only deprecated).

> - Declares C++17 polyfill implementations for ``optional<T>`` and ``string_view`` in the ``bsoncxx::v1::stdx`` namespace. The

Reviewer: Suggest moving this point into a new section "ABI Breaking Changes in v4.1".

Author: Moved to the breaking changes section of the upgrade page - linked in the warning box above

  ``bsoncxx::v_noabi::stdx`` namespace now contains type aliases to their v1
  equivalents.

  .. note::

     This does not affect users who compile with C++17 or newer and have not set ``BSONCXX_POLY_USE_IMPLS=ON``

- Adds support for the BSON Binary Vector type.  
  
  For more information, see the following resources:

  - :manual:`Binary Data BSON Types </reference/bson-types/#binary-data>` in the {+mdb-server+} manual
  - :github:`BSON Binary Vector </mongodb/mongo-cxx-driver/blob/master/examples/bsoncxx/bson_binary_vector.cpp>` usage example on GitHub

> - Fixes the :ref:`CMake API versioning error <v4.0-cmake-api-version-bug>` noted

Reviewer: ```suggestion
- Fixes the :ref:`C driver API version bug <v4.0-cmake-api-version-bug>` noted
```

Suggest referring to C driver, since this bug only applied to the auto-downloaded C driver (and not the C++ driver itself). I expect it will have low user impact.
